### PR TITLE
Add argument validation

### PR DESCRIPTION
**Pull request by BrennanConroy**
#795 
@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs [hunk 1]
```diff
@@ -164,6 +168,12 @@ public class OpenIdConnectMiddleware : AuthenticationMiddleware<OpenIdConnectOpt
                         new HttpDocumentRetriever(Backchannel) { RequireHttps = Options.RequireHttpsMetadata });
                 }
             }
+            
+            if (Options.ConfigurationManager == null)
+            {
+                throw new InvalidOperationException($"Provide {nameof(Options.Authority)}, {nameof(Options.MetadataAddress)}, "
+                + "{nameof(Options.Configuration)}, or {nameof(Options.ConfigurationManager)} to options");
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs]: Probably need $ here too?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs]: Yes

**Comment by Tratcher** [comment]
Negative tests?

**Comment by BrennanConroy** [comment]
Added some

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs [hunk 2]
```diff
@@ -82,6 +82,11 @@ public class OpenIdConnectMiddleware : AuthenticationMiddleware<OpenIdConnectOpt
                 throw new ArgumentNullException(nameof(htmlEncoder));
             }
 
+            if (string.IsNullOrEmpty(Options.ClientId))
+            {
+                throw new ArgumentException(nameof(Options.ClientId));
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs]: Wrong constructor. Use the two param one: https://msdn.microsoft.com/en-us/library/sxykka64(v=vs.110).aspx

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs]: With an actual message.

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**